sim/common/wotlk/shadowmourne.go

```diff
@@ -49,21 +49,22 @@ func init() {
 					return
 				}
 
+				if stackingAura.GetStacks() == 10 {
+					stackingAura.Deactivate(sim)
+					tempStrProc.Activate(sim)
+					choasBaneSpell.Cast(sim, player.CurrentTarget)
+				}
+
 				if sim.RandomFloat("shadowmourne") > drainChance {
 					return
 				}
 
 				if !stackingAura.IsActive() {
```

Comment:
Activate already checks for `IsActive`, so you don't need this check. Also, letting `Activate` do the check is better so it records the aura proc for metrics.

Reply:
ah, for some reason I thought activate would reset stacks. ill fix it